backend-rust/src/indexer.rs

```diff
@@ -2745,24 +2748,48 @@ async fn process_cis2_event(
                 .await
                 .context("Failed inserting or updating account balance from mint event")?;
             }
+
+            // Insert the token event into the table.
+            sqlx::query!(
+                "INSERT INTO cis2_token_events (
+                    index_per_token,
+                    transaction_index,
+                    token_index,
+                    cis2_token_event
+                )
+                SELECT
+                    COALESCE((SELECT MAX(index_per_token) + 1 FROM cis2_token_events), 0),
```

Comment:
It looks to be fetching the max across all tokens
```suggestion
                    COALESCE((SELECT MAX(index_per_token) + 1 FROM cis2_token_events WHERE cis2_token_events.token_index = tokens.index), 0),
```